src/Aeson.purs

```diff
@@ -379,6 +387,14 @@ instance (GDecodeAeson row list, RL.RowToList row list) => DecodeAeson (Record r
       Just object -> gDecodeAeson object (Proxy :: Proxy list)
       Nothing -> Left $ TypeMismatch "Object"
 
+else instance (InOneOf b a b, DecodeAeson a, DecodeAeson b) => DecodeAeson (a |+| b) where
+  decodeAeson j =
+     let a = asOneOf <$> (decodeAeson j :: Either JsonDecodeError a)
```

Comment:
Why not use `<|>`?

Reply:
I did start with (<|>) but then had some typechecking issues and forgot to fix it back.